### PR TITLE
Create from description impl

### PR DESCRIPTION
**Pull request by egekorkan**
I have implemented the createFromDescription for the Exposed thing. It works good from the tests I have made for my use case ( I haven't written any tests for it at the moment)

### CODE REVIEW

**Comment by egekorkan** [comment]
The problem with the HTTP post requests found: in case the mediaType was specifying the encoding or extra parameters, since the codecs don't this information, it would always see this mediaType as a non existing codec. I have written a method to isolate the first part of the supplied mediaType (until the semi column).

I am not sure if this is a clean solution but it works

File: packages/node-wot/src/wot-impl.ts [hunk 1]
```diff
@@ -104,26 +105,128 @@ export default class WoTImpl implements WoT.WoTFactory {
...
                 if (content.mediaType !== "application/json")
                     logger.warn(`WoTImpl parsing TD from '${content.mediaType}' media type`);
-                let thingdesc = TDParser.parseTDString(content.body.toString());
-                let mything = new ExposedThing(this.srv, thingdesc.name);
-                if (this.srv.addThing(mything)) {
-                    resolve(mything);
+                let thingDescription = TDParser.parseTDString(content.body.toString());
+                let myThing = new ExposedThing(this.srv, thingDescription.name);
+                if (this.srv.addThing(myThing)) {
+                    //copy paste from the createFromDescription
```
Comment (h0ru5) [inline packages/node-wot/src/wot-impl.ts]: could you call the method instead of copy-pasting this code?

File: examples/scripts/counter.js [hunk 2]
```diff
@@ -8,7 +8,7 @@
         .addProperty("count", { type: "integer" })
         .setProperty("count",0);
 	
-	thing
+	    thing
```
Comment (h0ru5) [inline examples/scripts/counter.js]: seems just a stray whitespace

File: README.md [hunk 3]
```diff
@@ -49,4 +49,12 @@ wot-servient
...
 * you can query the count by http://localhost:8080/counter/properties/count
 * you can modify the count via POST on http://localhost:8080/counter/actions/increment and http://localhost:8080/counter/actions/decrement
-* application logic is in ``examples/scripts/counter.js``
\ No newline at end of file
+* application logic is in ``examples/scripts/counter.js``
+
+## How to use the library
+
+This library implements the Scripting API defined in the [WoT Current Practices](https://w3c.github.io/wot/current-practices/wot-practices.html#scripting-api) document. 
+
```
Comment (h0ru5) [inline README.md]: We should also point to https://w3c.github.io/wot-scripting-api/ - but can do that when we merge tthe `new-api` branch

**Comment by egekorkan** [comment]
"minor import fix" fixed also the type declaration at line 108 "let thingDescription :_ThingDescription_= TDParser.parseTDString(content.body.toString());"

**Pull request merged**